cupboard_app/tests.py

```diff
@@ -414,6 +417,181 @@ def test_get_all_ingredients(self, mock_decode):
         )
 
 
+class AddIngredientToListApi(TestCase):
+    def setUp(self):
+        """
+        Sets up a test database with test values
+        """
+        User.objects.create(username="testuser", email="test@test.com")
+        ListName.objects.create(listName="testlist")
+        create_ingredient("test_ing", "test_type")
+        create_measurement("test_unit")
+        UserListIngredients.objects.create(
+            user=User.objects.get(username="testuser"),
+            listName=ListName.objects.get(listName="testlist"),
+            ingredients=[]
+        )
+        Ingredient.objects.create(name="testing2", type="TEST2")
+
+    @patch.object(TokenBackend, 'decode')
+    def test_add_ingredient_to_list(self, mock_decode):
+        """
+        Testing adding an ingredient to an existing list
+        """
+        mock_decode.return_value = TEST_VALID_TOKEN_PAYLOAD
+
+        response = self.client.post(
```

Comment:
Use a put request instead to follow restful standards and because this is technically updating a user's already existing list. In general, use put for updating and post for creating.
https://restfulapi.net/rest-put-vs-post/